### PR TITLE
fix signal issue (port of #63)

### PR DESCRIPTION
**Pull request by indam23**

### CODE REVIEW

**Comment by ancalita** [comment]
@melindaloubser1 Those extra 2 lines of code seem to cause some Sanic errors in `test_nlg` among others, I recommend removing them and keeping only the 1 line :
```
await self.app._startup()
```

File: pytest_sanic/utils.py [hunk 1]
```diff
@@ -100,6 +100,11 @@ async def start_server(self):
         server_settings.pop("main_start", None)
         server_settings.pop("main_stop", None)
 
+        await self.app._startup()
+        await self.app._server_event("init", "before")
+        # Trigger after_start events
+        await self.app._server_event("init", "after")
```
Comment (indam23) [inline pytest_sanic/utils.py]:
```suggestion
        await self.app._server_event("init", "before", loop=self.loop)
        await self.app._server_event("init", "after", loop=self.loop)
```

**Comment by indam23** [comment]
I've made one update according to [this comment](https://github.com/yunstanford/pytest-sanic/issues/61#issuecomment-1128068998), testing it now

**Comment by indam23** [comment]
With the loop specifies it works with the additions

**Comment by notzippy** [comment]
There is one more statement that needs to be called now with `Sanic 22.6` branch `app.prepare` I called it like this

         server_settings.pop("main_start", None)
        server_settings.pop("main_stop", None)

       # Added the prepare call with setting the state flage `self.app.asgi `
        if not self.app.state.server_info:
            self.app.prepare()
            # Alternately asgi_client = self.app.asgi_client
            self.app.asgi = True

        # We need to invoke the server directly
        await self.app._startup()
        await self.app._server_event("init", "before", loop=self.loop)

**Comment by indam23** [comment]
@notzippy If you want to take this over feel free, for our case we kept using our fork.

**Comment by notzippy** [comment]
> @notzippy If you want to take this over feel free, for our case we kept using our fork.

Ive kinda derived a custom version for my own from this repo, so I was just contributing back :-)

**Comment by yunstanford** [comment]
Hey guys @notzippy @melindaloubser1 , sorry about the late replies and inconvenience, kinda of busy these days, not able to keep maintaining the lib.

I'm happy to add you guys as maintainers (github and pypi) as well

**Comment by indam23** [comment]
all good, not needed in my case at least. I'll close this for now then.

**Comment by comino** [comment]
Why close this? This looks like good solution :) 
@yunstanford Can you consider merging this? Right now the plugin doesn't work with newer sanic versions.

**Pull request closed without merging**